tasks/measurements/ookla_speedtest_cli.py

```diff
@@ -0,0 +1,71 @@
+import time
+import subprocess
+from subprocess import Popen
+import os
+import logging
+import subprocess
+import time
+from netunicorn.base.architecture import Architecture # type: ignore
+from netunicorn.base.nodes import Node # type: ignore
+from netunicorn.base.task import Failure, Task, TaskDispatcher # type: ignore
+from netunicorn.client.remote import RemoteClient, RemoteClientException # type: ignore
+from netunicorn.base import Experiment, ExperimentStatus, Pipeline # type: ignore
+from netunicorn.library.tasks.basic import ShellCommand # type: ignore
+from netunicorn.base.architecture import Architecture # type: ignore
+from netunicorn.base.nodes import Node # type: ignore
+from netunicorn.base.task import Failure, Task, TaskDispatcher # type: ignore
+from netunicorn.base import Result, Failure, Success, Task, TaskDispatcher # type: ignore
+from netunicorn.base.architecture import Architecture # type: ignore
+from netunicorn.base.nodes import Node # type: ignore
+from returns.pipeline import is_successful # type: ignore
+from returns.result import Failure # type: ignore
+
+class Ookla_Speedtest_CLI(TaskDispatcher):
+    def __init__(self, filepath: str, server: str = '', ip: str = '', timeout: int = 120, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.linux_implementation = Ookla_Speedtest_CLI_LinuxImplementation(
+            filepath=filepath, timeout=timeout, server=server, ip=ip, name=self.name
+        )
+
+    def dispatch(self, node: Node) -> Task:
+        if node.architecture in {Architecture.LINUX_AMD64, Architecture.LINUX_ARM64}:
+            return self.linux_implementation
+        raise NotImplementedError(
+            f"Ookla_Speedtest_CLI is not implemented for architecture: {node.architecture}"
+        )
+    
+class Ookla_Speedtest_CLI_LinuxImplementation(Task):
+    def __init__(self, filepath: str, server: str = '', ip: str = '', timeout: int = 120, *args, **kwargs):
+        self.filepath = filepath
+        self.timeout = timeout
+        self.server = server
+        self.ip = ip
+        super().__init__(*args, **kwargs)
+    
+    def run(self):
+        try:
+            with open(self.filepath, 'w') as outfile:
```

Comment:
I believe you can avoid creating a new file and instead forward results directly from stdout to the python variable. Creating a file might fail on read-only file system or when multiple tasks are running in parallel, and also is not really something user should think about by default.